lib/src/view/game/game_result_dialog.dart

```diff
@@ -98,6 +99,62 @@ class _GameEndDialogState extends ConsumerState<GameResultDialog> {
           padding: const EdgeInsets.only(bottom: 16.0),
           child: GameResult(game: gameState.game),
         ),
+        AnimatedSize(
+          duration: const Duration(milliseconds: 400),
+          onEnd: () {
+            animationFinished.value = true;
+          },
+          child: ValueListenableBuilder<bool>(
+            valueListenable: animationFinished,
+            builder: (context, value, child) {
+              return Visibility(
+                maintainAnimation: true,
+                maintainSize: true,
+                maintainState: true,
+                visible: animationFinished.value,
+                child: Column(
+                  children: [
+                    if (gameState.game.opponent?.offeringRematch == true)
```

Comment:
Instead of returning an empty column if `gameState.game.opponent?.offeringRematch` is false, you could use a ternary operator to return either the column or a `Sizebox.shrink()` if the opponent is not offering rematch.